src/lib/mdns/Resolver.h

```diff
@@ -100,6 +103,66 @@ struct DiscoveredNodeData
     DiscoveredNodeData() { Reset(); }
     bool IsHost(const char * host) const { return strcmp(host, hostName) == 0; }
     bool IsValid() const { return !IsHost("") && ipAddress[0] != chip::Inet::IPAddress::Any; }
+    void Log() const
+    {
+        if (rotatingIdLen > 0)
+        {
+            char rotatingIdString[chip::Mdns::kMaxRotatingIdLen * 2 + 1] = "";
+            Encoding::BytesToUppercaseHexString(rotatingId, rotatingIdLen, rotatingIdString, sizeof(rotatingIdString));
+            ChipLogDetail(DiscoveryVerbose, "Rotating ID: %s", rotatingIdString);
+        }
```

Comment:
This must be conditional on the rotating ID feature (which is not always implemented)

Reply:
The generation and advertisement of a rotating ID is already conditional on the server side on #if CHIP_ENABLE_ROTATING_DEVICE_ID. The client side code here is already checking if no rotating ID was received by checking for rotatingIdLen > 0. Then, why must we put a #if CHIP_ENABLE_ROTATING_DEVICE_ID check here in the client?

Reply:
Because rotating ID is a fully optional feature. Try to turn off CHIP_ENABLE_ROTATING_DEVICE_ID and it should build. If it builds cleanly without additional `#ifdef` after `CHIP_ENABLE_ROTATING_DEVICE_ID` is disabled, then you can resolve this comment.

Reply:
Sure. I defined CHIP_ENABLE_ROTATING_DEVICE_ID and set it to 0 in the tv-casting-app's CHIPProjectAppConfig.h file. The build still succeeded.